src/Consumer.cc

```diff
@@ -82,9 +96,15 @@ void MessageListener(pulsar_consumer_t *rawConsumer, pulsar_message_t *rawMessag
     return;
   }
 
-  MessageListenerProxyData *dataPtr = new MessageListenerProxyData(cMessage, consumer);
+  std::promise<void> promise;
+  std::future<void> future = promise.get_future();
+  MessageListenerProxyData *dataPtr =
+      new MessageListenerProxyData(cMessage, consumer, [&]() { promise.set_value(); });
```

Comment:
```suggestion
      new MessageListenerProxyData(cMessage, consumer, [&promise]() { promise.set_value(); });
```

Capture the specific value or reference instead of capturing all values or references.